pkg/controller/nodenetworkconfigurationpolicy/conditions/enactments.go

```diff
@@ -0,0 +1,87 @@
+package conditions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+func setEnactmentFailedToConfigure(enactments *nmstatev1alpha1.EnactmentList, nodeName string, message string) {
+	setEnactmentFailed(enactments, nodeName, nmstatev1alpha1.NodeNetworkConfigurationEnactmentConditionFailedToConfigure, message)
+}
+
+func setEnactmentFailedToFindPolicy(enactments *nmstatev1alpha1.EnactmentList, nodeName string, message string) {
+	setEnactmentFailed(enactments, nodeName, nmstatev1alpha1.NodeNetworkConfigurationEnactmentConditionFailedToFindPolicy, message)
+}
+
+func setEnactmentFailed(enactments *nmstatev1alpha1.EnactmentList, nodeName string, reason nmstatev1alpha1.ConditionReason, message string) {
+	enactments.SetCondition(
```

Comment:
is it ok that this 3 commands aren't part of one critical section? 
or there isnt parallel execution that might be corrupted due to this ?

Reply:
we just do a retryOnConflict with this, there is no parallel code running.